### PR TITLE
Docstring

### PR DESCRIPTION
**Pull request by thulasiramanmt**
# Pull request

**Resolves:** #36

### CODE REVIEW

**Comment by DoodlePooge** [comment]
What is this PR for?

**Comment by thulasiramanmt** [comment]
> What is this PR for?
it's for #36

**Review by pelavarthi — DISMISSED** [review]
reviewed and it looks okay to me

**Review by Viraj-Ganguli — COMMENTED** [review]
Still looks good after merging with main!

**Comment by pelavarthi** [comment]
looks okay to me

**Review by Viraj-Ganguli — APPROVED** [review]
Looks good!

File: darkmoon/api/v1/metadata/views.py [hunk 1]
```diff
@@ -54,17 +50,30 @@ async def list_metadata(
...
+
+    Raises:
+        IncorrectInputException:
+            The input contains invalid UTF-8 characters
+            or is all white space
+        ServerNotFoundException:
+            Endpoint is unable to connect to mongoDB instance
 
     Raises:
         IncorrectInputException:
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py:79]: Raises Raises Raises...

File: darkmoon/app.py [hunk 2]
```diff
@@ -4,18 +4,43 @@
...
+
+    Parameters:
+        request (Request): The incoming request.
+        id (str): The ID of the item.
+
+    Returns:
+        Response: HTML response with the item.html template.
+    """
+    return templates.TemplateResponse("item.html", {"request": request, "id": id})
+
```
Comment (DoodlePooge) [inline darkmoon/app.py:33]: No.

**Review by DoodlePooge — CHANGES_REQUESTED** [review]

File: README.md
Comment (DoodlePooge) [inline README.md:1]: This is not the readme PR

File: darkmoon/app.py (hunk 2 above)
Comment (DoodlePooge) [reply darkmoon/app.py:33]: This was not resolved. Jinja logic is still here...

File: darkmoon/app.py [hunk 3]
```diff
@@ -4,18 +4,43 @@
...
 
 def get_app(settings: Settings | None = None) -> FastAPI:
-    """Return the FastAPI connection."""
-    app_settings = settings or Settings()
+    """Return the FastAPI connection.
 
-    app = FastAPI()
```
Comment (DoodlePooge) [inline darkmoon/app.py:18]: Why are you removing code?

**Review by DoodlePooge — CHANGES_REQUESTED** [review]

**Review by Viraj-Ganguli — APPROVED** [review]
Looks good!

**Review by Viraj-Ganguli — COMMENTED** [review]
Looks good!

File: darkmoon/api/v1/metadata/views.py [hunk 4]
```diff
@@ -54,25 +50,29 @@ async def list_metadata(
...
+        page (int): The page number to iterate to.
+        length (int): The number of items per page.
 
-    Returns:
-        documents (list[MetadataEntity]): List of all documents that match
-            parameters in the database
 
     Raises:
+        HTTPException: If the hash or hash_type is missing.
+        errors.ServerSelectionTimeoutError: If the server is not found.
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py:65]: Not being raised anymore

**Review by DoodlePooge — CHANGES_REQUESTED** [review]

**Review by Viraj-Ganguli — COMMENTED** [review]
Looks good!

File: README.md [hunk 5]
```diff
@@ -15,13 +15,10 @@ The Web API service will utilize FastAPI. This web service will be un-authentica
 
 -[FastAPI](https://fastapi.tiangolo.com)
 ### Pre-Commit Hooks
-
 To ensure consistent and high-quality code that follows standard practices, this repository will use pre-commit hooks. These checks must be passed before code will be able to be pushed to the repository.
 
 ### Conventional Commits
-
 To keep a easily readable and clean commit history, conventional commits will be enforced.
-
```
Comment (DoodlePooge) [inline README.md:24]: Please put these lines back

File: darkmoon/api/v1/metadata/views.py [hunk 6]
```diff
@@ -219,7 +216,6 @@ async def upload_metadata(
             uses invalid characters
         ServerNotFoundException:
             Endpoint is unable to connect to mongoDB instance
```
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py:218]: This should be under a raises section along with the other exceptions for this endpoint

File: darkmoon/api/v1/metadata/views.py (hunk 4 above)
Comment (DoodlePooge) [reply darkmoon/api/v1/metadata/views.py:65]: This was not resolved, why are we resolving issues that are not resolved? HTTPException is not called in this function anymore.

File: darkmoon/api/v1/metadata/views.py
Comment (DoodlePooge) [inline darkmoon/api/v1/metadata/views.py:1]: What is going on with that CLASSES comment block? remove it or fix it.

Comment (DoodlePooge) [reply darkmoon/api/v1/metadata/views.py:1]: Also for my sanity's sake, keep all your docstrings in the same order, pick either Parameters -> Raises -> Returns OR Parameters -> Returns -> Raises
Stop doing both orderings

**Review by DoodlePooge — CHANGES_REQUESTED** [review]

**Review by Viraj-Ganguli — COMMENTED** [review]
Looks good!

**Review by DoodlePooge — APPROVED** [review]

**Pull request merged**